pkg/queue/manager_test.go

```diff
@@ -478,9 +478,7 @@ func TestHeadsAsync(t *testing.T) {
 		if err := manager.AddQueue(ctx, &q); err != nil {
 			t.Errorf("Failed adding queue: %s", err)
 		}
-		go func() {
-			manager.addOrUpdateWorkload(&wl)
```

Comment:
This was intentional, but it should have been `manager.AddOrUpdateWorkload`, which takes the lock.

Reply:
Leave the `go` call please

Reply:
done